src/base/logging.hh

```diff
@@ -123,19 +126,44 @@ class Logger
      */
     [[noreturn]] void exit_helper() { exit(); ::abort(); }
 
+    /** Register callback to generate extra log. */
+    void
+    registerExtraLog(std::function<std::string()> cb)
+    {
+        extraLogs.emplace_back(std::move(cb));
+    }
+
   protected:
     bool enabled;
 
+    /** Generate the formatted extra log string. */
+    std::string
+    getFormattedExtraLog()
```

Comment:
Sorry, just noticed this. This should be a const function, as it is not expected to modify the class.

Reply:
Added const. Thanks for catching!